pkg/controller/v1alpha1/localmodel/controller.go

```diff
@@ -0,0 +1,643 @@
+			c.Log.Error(err, "Create pv err", "name", pv.Name)
+		}
+
+		pvc := v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      localModel.Name,
+				Namespace: namespace,
+			},
+			Spec: nodeGroup.Spec.PersistentVolumeClaimSpec,
+		}
+		pvc.Spec.VolumeName = pv.Name
+		c.Log.Info("Checking PVC")
+		if err := c.createPVC(pvc, namespace, localModel); err != nil {
+			c.Log.Error(err, "Create pvc err", "name", pvc.Name)
+		}
+	}
+	return nil
+}
+
+// Step 1 - Checks if the CR is in the deletion process, if so, it creates deletion jobs to delete models on all nodes.
+// Step 2 - Creates PV & PVC for model download
+// Step 3 - Creates Jobs on all nodes to download models
+// Step 4 - Creates PV & PVCs for namespaces with isvcs using this cached model
+func (c *LocalModelReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	c.Log.Info("Reconcile() called for localmodel", "name", req.Name)
+	localModel := &v1alpha1api.ClusterLocalModel{}
+	if err := c.Get(ctx, req.NamespacedName, localModel); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	nodeGroup := &v1alpha1api.LocalModelNodeGroup{}
+	nodeGroupNamespacedName := types.NamespacedName{Name: localModel.Spec.NodeGroup}
+	if err := c.Get(ctx, nodeGroupNamespacedName, nodeGroup); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Step 1 - Checks if the CR is in the deletion process, if so, creates deletion jobs to delete models on all nodes.
+	if localModel.ObjectMeta.DeletionTimestamp.IsZero() {
+		// The object is not being deleted, so if it does not have our finalizer,
+		// then lets add the finalizer and update the object. This is equivalent
+		// registering our finalizer.
+		if !utils.Includes(localModel.ObjectMeta.Finalizers, finalizerName) {
+			patch := client.MergeFrom(localModel.DeepCopy())
+			localModel.ObjectMeta.Finalizers = append(localModel.ObjectMeta.Finalizers, finalizerName)
+			if err := c.Patch(context.Background(), localModel, patch); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		return c.deleteModelFromNodes(localModel)
+	}
+
+	// Step 2 - Creates PV & PVC for model download
+	pvSpec := nodeGroup.Spec.PersistentVolumeSpec
+	pv := v1.PersistentVolume{Spec: pvSpec, ObjectMeta: metav1.ObjectMeta{
+		Name: localModel.Name + "-download",
+	}}
+	if err := c.createPV(pv, localModel); err != nil {
+		c.Log.Error(err, "Create pv err", "name", pv.Name)
+	}
+
+	pvc := v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: localModel.Name,
+		},
+		Spec: nodeGroup.Spec.PersistentVolumeClaimSpec,
+	}
+	pvc.Spec.VolumeName = pv.Name
+
+	if err := c.createPVC(pvc, jobNamespace, localModel); err != nil {
+		c.Log.Error(err, "Create pvc err", "name", pv.Name)
+	}
+
+	// Step 3 - Creates Jobs on all nodes to download models
+	if err := c.DownloadModel(ctx, localModel, nodeGroup, pvc); err != nil {
+		c.Log.Error(err, "Model download err", "model", localModel.Name)
```

Comment:
update logging to "failed to create model downloading jobs"

Reply:
It could be other errors from `DownloadModel`.